homeassistant/components/tplink/common.py

```diff
@@ -133,7 +133,7 @@ def get_static_devices(config_data) -> SmartDevices:
     async_add_entities: Callable[[List[Any], bool], None],
     objects: List[Any],
     callback: Callable[[Any, Callable], None],
-    interval: timedelta = timedelta(seconds=60),
+    interval: timedelta = timedelta(seconds=2),
```

Comment:
How will the short retry impact the device? Can it handle it?

Reply:
As far as I can tell leaving higher means that it gets added to HA without a name (shows up as unnamed). With the lower interval it seems to add everything fine, but I can't tell if it is coincidence or if something else is going on.

Reply:
There is something else wrong in there; the `async_add_entities` is called with `update_before_adding`, so the name should be known when the entity gets added.

Reply:
As far as I can tell, even with all of the devices I have, I don't see that retry occurring anyway because it always adds the devices on the first go, it is just that sometimes it gets no name back and then ends up adding the devices as unnamed (it is especially worse on the HS300 (the 6 port device), but I have seen it on some of the light switches as well). 

I am really just trying to get the "alias" and "device_id" values to be updated before add and I was trying to figure out if that was possible from just what Discover.discover() returns (which I assume would lessen the number of I/O requests).

Reply:
That information is part of the sysinfo response, see https://github.com/python-kasa/python-kasa/blob/master/kasa/tests/fixtures/HS300(US)_1.0.json for a dummy example. The problem is that for multi socket devices each socket is queried separately, and some devices simply start ignoring incoming packets if you send them too many in a short period of time.

We are currently moving the library to a new home, and cleaning up the code base as well as adding caching and async interfaces to it. The caching will likely fix the problem for you.

Reply:
Yeah, I just want to get something in temporarily because the current process is flat out not working especially if you have a lot of devices. I moved the other potential I/O back to the sync task (attempt_update). So I think at least for now this code is checking all the boxes until the other caching gets worked out.